### DISCUSSION TITLE
onSelect not working

### DISCUSSION THREAD

**Discussion by KingBier** [afconsult/quill-mention#316] [Q&A] [answered]
Hello,

I have the issue that the custom onSelect function is not working.
This are my mention module settings:
```
mention: {
			minChars: 2,
			spaceAfterInsert: false,
			showDenotationChar: false,
			isolateCharacter: true,
			dataAttributes: ['href'],
			allowedChars: /^[A-Za-z\sÅÄÖåÜäöü]*$/,
			mentionDenotationChars: ["@"],
			fixMentionsToQuill: true,
			source: async function (searchTerm, renderList) {
				const matchedTerms = await suggestTerms(searchTerm);
				renderList(matchedTerms);
			},
			onSelect: function (item, insertItem) {
				console.log(item);
				insertItem(item);
			}
		}
```

No console output is generated when I click on the suggestion, but the suggestion is inserted.
How can that be?
What am I missing?

**Comment by kotAPI** [comment]
I have a same issue on angularjs, It is working, but it doesn't always work for some reason. It works once in a while which makes it really tricky and hard to understand what's happening. Did you have any luck figuring this out?

**Comment by kotAPI** [comment, marked as answer]
> Hello,
> 
> I have the issue that the custom onSelect function is not working.
> This are my mention module settings:
> 
> ```
> mention: {
> 			minChars: 2,
> 			spaceAfterInsert: false,
> 			showDenotationChar: false,
> 			isolateCharacter: true,
> 			dataAttributes: ['href'],
> 			allowedChars: /^[A-Za-z\sÅÄÖåÜäöü]*$/,
> 			mentionDenotationChars: ["@"],
> 			fixMentionsToQuill: true,
> 			source: async function (searchTerm, renderList) {
> 				const matchedTerms = await suggestTerms(searchTerm);
> 				renderList(matchedTerms);
> 			},
> 			onSelect: function (item, insertItem) {
> 				console.log(item);
> 				insertItem(item);
> 			}
> 		}
> ```
> 
> No console output is generated when I click on the suggestion, but the suggestion is inserted.
> How can that be?
> What am I missing?

I seem to have fixed it by some css,setTimeout trickery. Its mostly happening because of angularjs's lifecycle.
